controller/src/controller.rs

```diff
@@ -168,6 +190,43 @@ impl<T: BottlerocketShadowClient> BrupopController<T> {
         ))
     }
 
+    #[instrument(skip(self, nodes, brss))]
+    async fn bottlerocketshadows_cleanup(
+        &self,
+        nodes: Vec<Node>,
+        brss: Vec<BottlerocketShadow>,
+    ) -> Result<()> {
+        let mut unlabeled_nodes = find_unlabeled_nodes(nodes);
+
+        for unlabeled_node in unlabeled_nodes.drain(..) {
+            let associated_bottlerocketshadow = brs_name_from_node_name(&unlabeled_node.name());
+            if brss
```

Comment:
I don't see any use for `BottlerocketShadow` when deleting the shadow. Does it make sense to get the BottlerocketShadow's name in `HashSet` to avoid iterating on all of the `brss` for each unlabeld_node?